server/src/main/java/org/elasticsearch/cluster/metadata/ComponentTemplate.java

```diff
@@ -103,11 +121,14 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(true);
             out.writeGenericMap(this.metadata);
         }
+        if (out.getTransportVersion().onOrAfter(TransportVersions.DEPRECATED_COMPONENT_TEMPLATES_ADDED)) {
+            out.writeOptionalBoolean(this.deprecated);
+        }
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(template, version, metadata);
+        return Objects.hash(template, version, metadata, deprecated());
```

Comment:
Same here about using
```suggestion
        return Objects.hash(template, version, metadata, deprecated);
```

So that we don't skip no-ops.